rest-server/src/main/scala/com/actionml/router/service/EngineService.scala

```diff
@@ -12,6 +15,30 @@ import scaldi.Injector
 
 trait EngineService extends ActorInjectable
 
+class CBEngineService(implicit inj: Injector) extends EngineService{
+
```

Comment:
The next version of pio-kappa should allow runtime addition of Templates so these Services will be configured by some parameters. Do nothing about this now, just FYI.

Some Datasets will invoke code from one Template and others code from another Template. These will always communicate with the "controller" which extends "Engine" but the instance may be different depending on the Template and the REST will be the same, though events will mean different things to different Templates and the same for queries. The REST will remain the same, what REST invokes will be different per Template.

Again, FYI only. Do nothing about this now.